mkat_fpga_tests/test_cbf.py

```diff
@@ -336,6 +333,91 @@ def print_baselines():
             fengops.feng_eq_set(self.correlator, source_name=inp, new_eq=old_eqs)
             zero_inputs.remove(inp)
             nonzero_inputs.add(inp)
-            #zero_baseline, nonzero_baseline = calc_zero_and_nonzero_baselines(nonzero_inputs)
-            #print_baselines()
-        #print self.correlator.feng_eq_get().items()
+
+    def test_back2back_consistency(self):
+        """1. Check that back-to-back dumps with same input are equal"""
+        test_name = '{}.{}'.format(strclass(self.__class__), self._testMethodName)
+        init_dsim_sources(self.dhost)
+        self.dhost.sine_sources.sin_0.set(frequency=self.expected_fc, scale=0.25)
+
+        for i, freq in enumerate(self.requested_test_freqs):
+            print ('Testing dump consistancy {}/{} @ {} MHz.'.format(
+                i+1, len(self.requested_test_freqs), freq/1e6))
+            self.dhost.sine_sources.sin_0.set(frequency=freq, scale=0.125)
+            dumps_data = []
+            for dump_no in range(3):
+                if dump_no == 0:
+                    this_freq_dump = self.receiver.get_clean_dump(DUMP_TIMEOUT)
+                else:
+                    this_freq_dump = self.receiver.data_queue.get(DUMP_TIMEOUT)
+                this_freq_data = this_freq_dump['xeng_raw']
+                max_freq_init_data = np.max(this_freq_dump['xeng_raw'])
+                dumps_data.append(this_freq_data)
+
+            for comparison in range(1, len(dumps_data)):
+                d0 = dumps_data[comparison - 1]
+                d1 = dumps_data[comparison]
+                diff_dumps = np.max(d0 - d1)
+
+            self.assertLess((diff_dumps/max_freq_init_data), self.threshold,
+                    'dump comparison({}) is >= {} self.threshold.'
+                        .format((diff_dumps/max_freq_init_data), self.threshold))
+
+    def test_freq_scan_consistency(self):
+        """2. Check that identical frequency scans produce equal results"""
+        test_name = '{}.{}'.format(strclass(self.__class__), self._testMethodName)
+        init_dsim_sources(self.dhost)
+        self.dhost.sine_sources.sin_0.set(frequency=self.expected_fc, scale=0.25)
+
+        scans = []
+        for scan_i in range(3):
+            scan_dumps = []
+            scans.append(scan_dumps)
+            for i, freq in enumerate(self.requested_test_freqs):
+                print ('Testing frequency scan consistancy {}/{} @ {} MHz.'.format(
+                i+1, len(self.requested_test_freqs), freq/1e6))
+                self.dhost.sine_sources.sin_0.set(frequency=freq, scale=0.125)
+                this_freq_dump = self.receiver.get_clean_dump(DUMP_TIMEOUT)
+                this_freq_data = this_freq_dump['xeng_raw']
+                max_freq_init_data = np.max(this_freq_dump['xeng_raw'])
```

Comment:
You only want to do this on the first scan, so put a 'if scan_i == 0' here. Similar comments apply to the other consistency tests.